jwst/assign_wcs/tests/test_nirspec.py

```diff
@@ -1182,3 +1178,93 @@ def test_in_slice(slice, wcs_ifu_grating, ifu_world_coord):
 
     r, d, _ = slice_wcs(x, y)
     assert r[~np.isnan(r)].size == xinv.size
+
+
+@pytest.mark.parametrize("mode", ["IFU", "MOS", "FS"])
+def test_nrs_wcs_by_slit(mode):
+    pixel_tol = 1e-7
+    if mode == "IFU":
+        hdul = create_nirspec_ifu_file("F290LP", "G140M")
+        im = datamodels.IFUImageModel(hdul)
+
+        # Round trip is currently up to ~half pixel off for IFU
+        pixel_tol = 0.5
+
+    elif mode == "MOS":
+        hdul = create_nirspec_mos_file()
+        im = datamodels.ImageModel(hdul)
+        msaconfl = get_pkg_data_filename(
+            "data/msa_configuration.fits", package="jwst.assign_wcs.tests")
+        im.meta.instrument.msa_metadata_file = msaconfl
+        im.meta.instrument.msa_metadata_id = 12
+    else:
+        hdul = create_nirspec_fs_file(grating="G140M", filter="F100LP")
+        im = datamodels.ImageModel(hdul)
+
+    datamodel = assign_wcs_step.AssignWcsStep.call(im)
+    slit_ids = datamodel.meta.wcs.get_transform('gwa', 'slit_frame').slit_ids
+    for slit_id in slit_ids:
+        x, y = wcstools.grid_from_bounding_box(datamodel.meta.wcs.bounding_box[slit_id])
+        ra, dec, lam, slit = datamodel.meta.wcs(x, y, slit_id)
+        is_nan = np.isnan(ra) | np.isnan(dec) | np.isnan(lam)
+
+        # Make sure the return values are nontrivial
+        assert np.any(~is_nan)
+
+        # Currently the returned slit value is sometimes NaN when called in this manner
+        # This test ought to work, but may need a gwcs bug fix.
+        # assert int(slit) == slit_id
+
+        # Check roundtrip
+        inv_x, inv_y, inv_slit = datamodel.meta.wcs.backward_transform(ra, dec, lam, slit_id)
+        assert np.allclose(inv_x[~is_nan], x[~is_nan], atol=pixel_tol)
```

Comment:
Why not use `numpy.testing.assert_allclose`? Same comment applies to all similar occurrences in the diff.

```suggestion
        assert_allclose(inv_x[~is_nan], x[~is_nan], atol=pixel_tol)
```

Reply:
No reason, I'm just used to using np.allclose.  I'll switch over.